Bug 2018481: OpenStack: Add note about ETP=Local and Octavia (#46245)
openshift/openshift-docs: modules/installation-osp-setting-cloud-provider-options.adoc

[LoadBalancer]
use-octavia=true <1>
lb-provider = "amphora" <2>
floating-network-id="d3deb660-4190-40a3-91f1-37326fe6ec4a" <3>
create-monitor = True <4>
monitor-delay = 10s <5>
monitor-timeout = 10s <6>
monitor-max-retries = 1 <7>
#...
----
<1> This property enables Octavia integration.
<2> This property sets the Octavia provider that your load balancer uses. It accepts `"ovn"` or `"amphora"` as values. If you choose to use OVN, you must also set `lb-method` to `SOURCE_IP_PORT`.
<3> This property is required if you want to use multiple external networks with your cluster. The cloud provider creates floating IP addresses on the network that is specified here.
<4> Setting this property to `True` will make cloud provider create health monitors for the Octavia load balancers. As of {rh-openstack} 16.x this is only supported for Amphora provider.
<5> This property is required when `create-monitor` is set to `True`. It controls how often endpoints are monitored.

> <6> This property is required when `create-monitor` is set to `True`. It controls timeout of the monitor requests.

Reviewer: ```suggestion
<6> This property is required when `create-monitor` is set to `True`. It sets how long monitoring requests take to time out in seconds. 
```

> <6> This property is required when `create-monitor` is set to `True`. It sets how long monitoring requests time out in `time.ParseDuration()` format.

Reviewer 2: Not sure this is grammatically correct.
```suggestion
<6> This property is required when `create-monitor` is set to `True`. It sets how long before monitoring requests time out in `time.ParseDuration()` format.
```

Reviewer: Best way to ensure grammatical correctness is to start from a rudimentary but painfully wordy statement and then whittle it down. 

How would you explain what is happening here to someone who's coming into this cold?

<7> This property is required when `create-monitor` is set to `True`. It controls how many successful monitor requests are needed before load balancer member is marked as online.
+
[IMPORTANT]
====
Prior to saving your changes, verify that the file is structured correctly. Clusters might fail if properties are not placed in the appropriate section.
====
+
[IMPORTANT]
====

> Setting `create-monitor` to `True` is required when using services with `.spec.endpointTrafficPolicy` set to `Local`. Please note that OVN Octavia provider in {rh-openstack} 16.x does not support health monitors and therefore `ETP=Local` services will work incorrectly when `lb-provider` is set to `"ovn"`.

Author: I'll add more specific consequences.

[LoadBalancer]
use-octavia=true <1>
lb-provider = "amphora" <2>
floating-network-id="d3deb660-4190-40a3-91f1-37326fe6ec4a" <3>
create-monitor = True <4>
monitor-delay = 10s <5>
monitor-timeout = 10s <6>
monitor-max-retries = 1 <7>
#...
----
<1> This property enables Octavia integration.
<2> This property sets the Octavia provider that your load balancer uses. It accepts `"ovn"` or `"amphora"` as values. If you choose to use OVN, you must also set `lb-method` to `SOURCE_IP_PORT`.
<3> This property is required if you want to use multiple external networks with your cluster. The cloud provider creates floating IP addresses on the network that is specified here.

> <4> Setting this property to `True` causes the cloud provider to create health monitors for the Octavia load balancers. As of {rh-openstack} 16.1 and 16.2 this is only supported for Amphora provider.
> <5> This property is required if `create-monitor` is set to `True`. It sets how often endpoints are monitored. The value must be in the `time.ParseDuration()` format.
> <6> This property is required if `create-monitor` is set to `True`. It sets the duration that monitoring requests remain open before timing out. The value must be in the `time.ParseDuration()` format.
> <7> This property is required if `create-monitor` is set to `True`. It controls how many successful monitor requests are needed before a load balancer member is marked as online.

Reviewer: ```suggestion
<4> This property controls whether the cloud provider creates health monitors for Octavia load balancers. Set the value to `True` to create health monitors. As of {rh-openstack} 16.1 and 16.2, this feature is only available for the Amphora provider.
<5> This property sets the frequency with which endpoints are monitored. The value must be in the `time.ParseDuration()` format. This property is required if the value of the `create-monitor` property is `True`.
<6> This property sets the time that monitoring requests are open before timing out. The value must be in the `time.ParseDuration()` format. This property is required if the value of the `create-monitor` property is `True`.
<7> This property defines how many successful monitoring requests are required before a load balancer is marked as online. The value must be an integer. This property is required if the value of the `create-monitor` property is `True`. 
```

Author: Done.

+
[IMPORTANT]
====
Prior to saving your changes, verify that the file is structured correctly. Clusters might fail if properties are not placed in the appropriate section.
====
+
[IMPORTANT]
====

> You must set `create-monitor` parameter to `True` if you use services that have `.spec.endpointTrafficPolicy` set to `Local`. The OVN Octavia provider in {rh-openstack} 16.1 and 16.2 does not support health monitors. Therefore, services that have `ETP` parameter values set to `Local` will randomly not respond when the `lb-provider` value is set to `"ovn"`.

Reviewer: ```suggestion
You must set the `create-monitor` parameter to `True` if you use services that have the `.spec.endpointTrafficPolicy` parameter set to `Local`. The OVN Octavia provider in {rh-openstack} 16.1 and 16.2 does not support health monitors. Therefore, services that have `ETP` parameter values set to `Local` will randomly not respond when the `lb-provider` value is set to `"ovn"`.
```

> You must set the `create-monitor` parameter to `True` if you use services that have the `.spec.endpointTrafficPolicy` parameter set to `Local`. The OVN Octavia provider in {rh-openstack} 16.1 and 16.2 does not support health monitors. Therefore, services that have `ETP` parameter values set to `Local` will randomly not respond when the `lb-provider` value is set to `"ovn"`.

Reviewer: To confirm--"parameter" or "property?" 

```suggestion
You must set the value of the `create-monitor` parameter to `True` if you use services that have the value of the `.spec.endpointTrafficPolicy` parameter set to `Local`. The OVN Octavia provider in {rh-openstack} 16.1 and 16.2 does not support health monitors. Therefore, services that have `ETP` parameter values set to `Local` might not respond when the `lb-provider` value is set to `"ovn"`.
```
